packages/vaadin-upload/src/vaadin-upload.js

```diff
@@ -512,6 +514,13 @@ class UploadElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     return maxFiles >= 0 && numFiles >= maxFiles;
   }
 
+  /** @private */
+  _maxFilesReachedChanged(maxFilesReached) {
+    maxFilesReached
+      ? this.setAttribute('max-files-reached', maxFilesReached)
```

Comment:
This would add `<vaadin-upload max-files-reached="true">` instead of `<vaadin-upload max-files-reached>`

Reply:
True, fixed by using `reflectToAttribute` instead of a custom observer.